lib/options.gi

```diff
@@ -15,8 +15,12 @@
 ## don't replace it, so we can make it Read Only
 ##
 
-OptionsStack := [];
-MakeReadOnlyGlobal("OptionsStack");
+if IsBound(HPCGAP) then
+    BindThreadLocal("OptionsStack", [ ]);
+else
+    OptionsStack := [];
+    MakeReadOnlyGlobal("OptionsStack");
```

Comment:
That's not quite right, the `MakeReadOnlyGlobal("OptionsStack");`  needs to be after the `fi;`. Because `BindThreadLocal` somewhat (?) confusingly does not invoke it, unlike `BindGlobal`.